Include/internal/pycore_freelist.h

```diff
@@ -147,6 +156,39 @@ extern void _PyAsyncGen_ClearFreeLists(struct _Py_object_freelists *freelists, i
 extern void _PyContext_ClearFreeList(struct _Py_object_freelists *freelists, int is_finalization);
 extern void _PyObjectStackChunk_ClearFreeList(struct _Py_object_freelists *freelists, int is_finalization);
 
+// Keep in sync with _asynciomodule.c !
+typedef struct futureiterobject_dummy {
+    PyObject_HEAD
+    void *future;
+} futureiterobject_dummy;
+
+static inline void
+_PyAsyncModule_ClearFreeLists(struct _Py_object_freelists *freelists, int is_finalization)
+{
+#ifdef WITH_FREELISTS
```

Comment:
Does anyone with better freelist internals knowledge know why this happens:

When I remove this function's body, the refleak tests pass. Otherwise they currently fail.